tests/test_rss.py

```diff
@@ -0,0 +1,266 @@
+                               responses=[
+                                    httpretty.Response(body=request_callback) \
+                                    for _ in range(2)
+                               ])
+
+    @httpretty.activate
+    def test_fetch(self):
+        """Test whether a list of entries is returned"""
+
+        self.__configure_http_server()
+
+        # Test fetch entries from feed
+        rss = RSS(RSS_FEED_URL)
+        entries = [item for item in rss.fetch()]
+        self.assertEqual(len(entries), 30)
+
+        # Test metadata
+        expected = [('98572defb3a652afbfdfe96517edefb88a22dcfa', 1481044620.0),
+                    ('e3b0d7463fca0c47d82debc3ddc37d8906f77548', 1480955040.0),
+                    ('28bca39353ff0825f53b157b69da319e2f49ab4d', 1480886040.0)]
+
+        for x in range(len(expected)):
+            item = entries[x]
+            self.assertEqual(item['origin'], 'http://example.com/rss')
+            self.assertEqual(item['uuid'], expected[x][0])
+            self.assertEqual(item['updated_on'], expected[x][1])
+            self.assertEqual(item['category'], 'item')
+            self.assertEqual(item['tag'], 'http://example.com/rss')
+
+    @httpretty.activate
+    def test_fetch_empty(self):
+        """Test whether it works when no entries are fetched"""
+
+        body = """
+        <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+        <rss version="2.0">
+        </rss>
+        """
+        httpretty.register_uri(httpretty.GET,
+                               RSS_FEED_URL,
+                               body=body, status=200)
+
+        rss = RSS(RSS_FEED_URL)
+        entries = [item for item in rss.fetch()]
+
+        self.assertEqual(len(entries), 0)
+
+
+class TestRSSBackendCache(unittest.TestCase):
+    """RSS backend tests using a cache"""
+
+    def setUp(self):
+        self.tmp_path = tempfile.mkdtemp(prefix='perceval_')
+
+    def tearDown(self):
+        shutil.rmtree(self.tmp_path)
+
+    @httpretty.activate
+    def test_fetch_from_cache(self):
+        """Test whether the cache works"""
+
+        bodies_entries = read_file('data/rss_entries.xml')
+
+        def request_callback(method, uri, headers):
+            if uri.startswith(RSS_FEED_URL):
+                body = bodies_entries
+            else:
+                body = ''
+
+            return (200, headers, body)
+
+        httpretty.register_uri(httpretty.GET,
+                               RSS_FEED_URL,
+                               responses=[
+                                    httpretty.Response(body=request_callback) \
+                                    for _ in range(2)
+                               ])
+
+        # First, we fetch the entries from the server, storing them
+        # in a cache
+        cache = Cache(self.tmp_path)
+        rss = RSS(RSS_FEED_URL, cache=cache)
+
+        entries = [item for item in rss.fetch()]
+
+        # Now, we get the entries from the cache.
```

Comment:
You should check the number of requests here, to know whether new ones were sent/received or not.